### PR TITLE
[iot] Improve documentation organization and add new material

### PR DESCRIPTION
**Pull request by dejanb**
### Type of change

- Documentation

### Description

This PR contains the following changes:

- Restructuring of the docs according to the user personas
- Adding examples for Commands using HTTP and MQTT
- Adding an examples of IoT configuration (from #3603)

### CODE REVIEW

**Comment by jbtrystram** [comment]
@dejanb You might want to pull the iot config examples I added here : https://github.com/jbtrystram/enmasse/pull/7

I also fixed an issues with some links that were not working properly

File: documentation/modules/proc-iot-receiving-commands-http.adoc [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// assembly-iot-solution-developers-guide.adoc
+
+[id='proc-iot-receiving-commands-http-{context}']
+= Receiving commands using HTTP protocol
+
+You can send commands from the cloud to a device using the HTTP protocol.
+
+.Procedure
+
+. Send telemetry message using the HTTP protocol, specifying `hono-ttd` parameter indicating for how long the client will wait for the command:
```
Comment (jenmalloy) [inline documentation/modules/proc-iot-receiving-commands-http.adoc]:
```suggestion
. Send a telemetry message using the HTTP protocol, specifying the `hono-ttd` parameter indicating how long the client will wait for the command:
```

File: documentation/modules/proc-iot-receiving-commands-http.adoc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
++
+[options="nowrap",subs="attributes"]
+----
+curl --insecure -X POST -i -u sensor1@myapp.iot:hono-secret -H 'Content-Type: application/json' --data-binary '{"temp": 5}' https://{http-adapter}/telemetry?hono-ttd=600
+----
+include::../common/snip-iot-customer-command.adoc[]
+. The client should receive command in the HTTP response
```
Comment (jenmalloy) [inline documentation/modules/proc-iot-receiving-commands-http.adoc]:
```suggestion
The client receives the command in the HTTP response:
```

File: documentation/modules/proc-iot-receiving-commands-mqtt.adoc [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// Module included in the following assemblies:
+//
+// assembly-iot-tenant-guide.adoc
+// assembly-iot-solution-developers-guide.adoc
+
+[id='proc-iot-receiving-commands-mqtt-{context}']
+= Receiving commands using MQTT protocol
```
Comment (jenmalloy) [inline documentation/modules/proc-iot-receiving-commands-mqtt.adoc]:
```suggestion
= Receiving commands using the MQTT protocol
```

**Comment by dejanb** [comment]
@jbtrystram I chery-picked fix link commit. Thanks. Let's put "IoT projects configuration examples" under the new top-level section "IoT for messaging tenants" and create a new PR for that

File: documentation/modules/proc-iot-receiving-commands-mqtt.adoc [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+You can send commands from the cloud to a device using the MQTT protocol.
+
+.Procedure
+
+. Use MQTT client to subscribe to the MQTT topic for receiving commands:
```
Comment (jenmalloy) [inline documentation/modules/proc-iot-receiving-commands-mqtt.adoc]:
```suggestion
. Use the MQTT client to subscribe to the MQTT topic for receiving commands:
```

File: documentation/modules/proc-iot-receiving-commands-mqtt.adoc [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
++
+[options="nowrap",subs="attributes"]
+----
+mosquitto_sub -v -d -h $(oc -n enmasse-infra get routes iot-mqtt-adapter --template='{{ .spec.host }}') -p 443 -u 'sensor1@myapp.iot' -P hono-secret -t command/+/+/req/# -i 4711 --cafile install/components/iot/examples/k8s-tls/build/iot-mqtt-adapter-fullchain.pem
+----
+include::../common/snip-iot-customer-command.adoc[]
+. The client should receive the command in the MQTT message
```
Comment (jenmalloy) [inline documentation/modules/proc-iot-receiving-commands-mqtt.adoc]:
```suggestion
The client receives the command in the MQTT message:
```

File: documentation/modules/ref-iot-config-example-adapters.adoc [hunk 6]
```diff
@@ -0,0 +1,17 @@
+// Module included in the following assemblies:
+//
+// assembly-iot-service-admin-guide.adoc
+
+[id='iot-adapters-example-{context}']
+= Tuning IoT protocol adapters example
```
Comment (jenmalloy) [inline documentation/modules/ref-iot-config-example-adapters.adoc]:
```suggestion
= Tuning the IoT protocol adapters example
```

File: documentation/modules/ref-iot-config-example-adapters.adoc [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+This IoT configuration example shows how the protocol adapters can be individually tuned.
+
+[source,yaml,options="nowrap"]
+----
+include::../common/iot-config-example1.yaml[]
+----
+<1> Protocol adapters can be disabled if necessary. The default value is `true`.
+<2> The duration for which client connection are kept alive for.
+<3> The maximum allowed size of an incoming message in bytes.
```
Comment (jenmalloy) [inline documentation/modules/ref-iot-config-example-adapters.adoc]:
```suggestion
<3> Specifies the maximum allowed size of an incoming message in bytes.
```

**Pull request merged**